src/kohlrahbi/__init__.py

```diff
@@ -21,12 +21,33 @@
 _pruefi_pattern = re.compile(r"^[1-9]\d{4}$")
 
 
-def get_valid_pruefis(list_of_pruefis: list[str]) -> list[str]:
+def get_valid_pruefis(list_of_pruefis: list[str], all_known_pruefis: Optional[list[str]] = None) -> list[str]:
     """
     This function returns a new list with only those pruefis which match the pruefi_pattern.
+    It also supports '*' style wildcards iff a list of known pruefis is given.
+    E.g. '11*' for all pruefis starting with '11' or '*01' for all pruefis ending with '01'.
     """
-    valid_pruefis: list[str] = [pruefi for pruefi in list_of_pruefis if _pruefi_pattern.match(pruefi)]
-    return valid_pruefis
+    result: set[str] = set()
+    for pruefi in list_of_pruefis:
+        # We could replace every occurrence of "*" with ".*" and try to interpret the result as a regex
+        # But do we want this? Do we need it? The following branches are all covered by unittests.
+        # We can still add the more complex regex support once there's the need for it.
+        if pruefi.count("*") > 1:
+            raise ValueError(f"Only one wildcard '*' is allowed per pruefi: '{pruefi}'")
+        if pruefi.endswith("*") and all_known_pruefis:
+            result = result.union({x for x in all_known_pruefis if x.startswith(pruefi[:-1])})
+        elif pruefi.startswith("*") and all_known_pruefis:
+            result = result.union({x for x in all_known_pruefis if x.endswith(pruefi[1:])})
+        elif "*" in pruefi and all_known_pruefis:
+            first_part = pruefi.split("*")[0]
+            second_part = pruefi.split("*")[1]
+            result = result.union(
+                {x for x in all_known_pruefis if x.startswith(first_part) and x.endswith(second_part)}
+            )
+        elif _pruefi_pattern.match(pruefi):
+            result.add(pruefi)
+        # else: is neither a wildcard nor a valid pruefi
```

Comment:
I would prefer the `fnmatch` package cause it will solve the same issue and more.
The expectation of the user would be, if I can use `*` then I can also use the `?` wildcard. With fnmatch we have exactly this behaviour which we know from other CLI tools